third_party/istio-1.6.1/generate.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generate manifest files from IstioOperator.
+istioctl manifest generate -f istio-ci-mesh-operator.yaml > istio-ci-mesh.yaml
```

Comment:
Yes, I think we can use `--set` option instead of `*-operator.yaml`.
But it makes command line option super long :cold_sweat: 

For example, I tried to write it for __only__ ingresses below. It looks like difficult to read/maintain it?

```
istioctl manifest generate                                                \
--set components.ingressGateways[0].enabled=true                          \
--set components.ingressGateways[0].k8s.resources.limits.cpu=3000m        \
--set components.ingressGateways[0].k8s.resources.limits.memory=2048Mi    \
--set components.ingressGateways[0].k8s.resources.requests.cpu=1000m      \
--set components.ingressGateways[0].k8s.resources.requests.memory=1024Mi  \
--set components.ingressGateways[1].enabled=true                          \
--set components.ingressGateways[1].name=my-cluster-local-gateway         \
--set components.ingressGateways[1].label.istio=cluster-local-gateway     \
--set components.ingressGateways[1].label.app=cluster-local-gateway       \
--set components.ingressGateways[1].k8s.resources.limits.cpu=3000m        \
--set components.ingressGateways[1].k8s.resources.limits.memory=2048Mi    \
--set components.ingressGateways[1].k8s.resources.requests.cpu=1000m      \
--set components.ingressGateways[1].k8s.resources.requests.memory=1024Mi  \
--set components.ingressGateways[1].k8s.service.type=ClusterIP            \
--set components.ingressGateways[1].k8s.service.ports[0].port=15020       \
--set components.ingressGateways[1].k8s.service.ports[0].name=status-port \
--set components.ingressGateways[1].k8s.service.ports[1].port=80          \
--set components.ingressGateways[1].k8s.service.ports[1].name=http2	  \
--set components.ingressGateways[1].k8s.service.ports[2].port=443	  \
--set components.ingressGateways[1].k8s.service.ports[2].name=https

# TODO: Add `spec.values`, `spec.mneshConfig`, `spec.addonComponents`.
```

Reply:
I think Nghia was just proposing to enable `PILOT_ENABLE_STATUS` and `enableAnalysis`, no specifically to use `--set`.
I am assuming it can be done with the operator CRD as well.